### PR TITLE
Fixes #31 : [Feat] Add a Sign-In button to sign in with Xee

### LINKED ISSUE #31: [Feat] Add a Sign-In button to sign in with Xee
It would be nice to add a Sign-In button to sign in with Xee easily

### PR DESCRIPTION
**Pull request by jcholin**
Sign-In button added to sign in with Xee !

### CODE REVIEW

File: sdk-api/src/main/java/com/xee/auth/SignInButton.java [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+import android.widget.Button;
+
+import com.xee.R;
+import com.xee.api.Xee;
+
+import java.lang.ref.WeakReference;
+
+public final class SignInButton extends Button {
+
+    private static final int TEXT_WHITE = 0xFFFFFFFF;
```
Comment (qk7b) [inline sdk-api/src/main/java/com/xee/auth/SignInButton.java]: This should be in the `res/color.xml` file.

File: README.md [hunk 2]
```diff
@@ -177,6 +177,50 @@ xeeApi.getTrips(carId, beginDate, endDate)
...
+You need to set a callback for result when clicking on the button by using the `setOnSignInClickResult` method and pass the `Xee API` instance that you have set (see [Setup](#setup) step)
+
+```java
+signInButton.setOnSignInClickResult(xeeApi, this);
+```
+
+Then implement the `ConnectionCallback` interface and override the `OnError` and `OnSuccess` methods to handle result
+
+```java
+public class MainActivity extends AppCompatActivity implements ConnectionCallback {
```
Comment (qk7b) [inline README.md]: Either you do that, or you can pass an anonymous implementation of `ConnectionCallback` in the method

File: sdk-api/src/main/java/com/xee/auth/SignInButton.java [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    private static final int TEXT_GREY = 0xFF474747;
```
Comment (qk7b) [inline sdk-api/src/main/java/com/xee/auth/SignInButton.java]: Same here

File: sdk-api/src/main/java/com/xee/auth/SignInButton.java [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+    private static final int SIZE_ICON = 24;
```
Comment (qk7b) [inline sdk-api/src/main/java/com/xee/auth/SignInButton.java]: In `res/dimen.xml` ?

File: sdk-api/src/main/java/com/xee/auth/SignInButton.java [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+        setBtnTheme(mTheme);
+
+        a.recycle();
+    }
+
+    /**
+     * Set the button size
+     * @param size the button size {@link Size}
+     */
+    public void setBtnSize(Size size) {
```
Comment (qk7b) [inline sdk-api/src/main/java/com/xee/auth/SignInButton.java]: Size should be `@NonNull`

File: sdk-api/src/main/java/com/xee/auth/SignInButton.java [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+
+        mDrawable.setBounds(0, 0, dpToPx(SIZE_ICON), dpToPx(SIZE_ICON));
+        setCompoundDrawables(null, null, mDrawable, null);
+    }
+
+    /**
+     * Set the button theme
+     * @param theme the button theme {@link Theme}
+     */
+    public void setBtnTheme(Theme theme) {
```
Comment (qk7b) [inline sdk-api/src/main/java/com/xee/auth/SignInButton.java]: Theme should be `@NonNull`

File: sdk-api/src/main/java/com/xee/auth/SignInButton.java [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+        mTheme = theme.ordinal();
+        switch (mTheme) {
+            case 0:
```
Comment (qk7b) [inline sdk-api/src/main/java/com/xee/auth/SignInButton.java]: Why don't you switch directly on the `theme` itself as it is an `enum` ?

**Comment by jcholin** [comment]
It closes #31

**Pull request merged**